src/utils/generateDirectionalStyles.js

```diff
@@ -0,0 +1,54 @@
+import rtlCSSJS from 'rtl-css-js';
+
+function isEmpty(obj) {
+  return Object.keys(obj).length === 0;
+}
+
+function separateDirectionalStyles(originalStyles, autoRTLStyles) {
+  const sharedStyles = {};
+  const ltrStyles = { ...originalStyles };
+  const rtlStyles = {};
+
+  let hasRTLStyles = false;
+  Object.entries(autoRTLStyles)
+    .forEach(([key, value]) => {
+      if (value === originalStyles[key]) {
+        delete ltrStyles[key];
+        sharedStyles[key] = value;
+        return;
+      }
+
+      if (value && typeof value === 'object') {
+        // In some cases (pseudoselectors, matchmedia queries, etc.), the style
+        // value may be an object, and we need to recurse.
+        const recursiveStyles = separateDirectionalStyles(originalStyles[key], value);
+        if (recursiveStyles != null) {
+          hasRTLStyles = true;
+          const {
+            sharedStyles: recursiveSharedStyles,
+            ltrStyles: recursiveLtrStyles,
+            rtlStyles: recursiveRtlStyles,
+          } = recursiveStyles;
+
+          if (!isEmpty(recursiveSharedStyles)) sharedStyles[key] = recursiveSharedStyles;
```

Comment:
is there a reason to check "isEmpty", as opposed to just setting it anyways, or merging it in?

Reply:
I am unsure. I can double-check to see how it behaves.